deepspeed/runtime/engine.py

```diff
@@ -640,10 +640,10 @@ def _configure_basic_optimizer(self, model_parameters):
 
         if self.optimizer_name() in [ADAM_OPTIMIZER, ADAMW_OPTIMIZER]:
             torch_adam = optimizer_parameters.pop(TORCH_ADAM_PARAM, False)
-            adam_w_mode = self.optimizer_name() == ADAMW_OPTIMIZER
+            adam_w_mode = optimizer_parameters.get(ADAM_W_MODE, ADAM_W_MODE_DEFAULT)
```

Comment:
I am a bit confused on why we have ADAMW_OPTIMIZER and ADAM_W_MODE both. What if the user passes adamw optimizer and sets adam_w_mode to false?

Reply:
Does it make sense to get rid of ADAMW_OPTIMIZER, and set ADAM_W_MODE to True by default?

Reply:
@samyam We added AdamW optimizer type last month because of https://github.com/microsoft/DeepSpeed/issues/668

Reply:
> What if the user passes adamw optimizer and sets adam_w_mode to false?

I do agree with @samyam about this. @jeffra maybe we want to assert adam_w_mode as true when ADAMW_OPTIMIZER is used?

Reply:
resolved offline with new structure